lib/grpnice.gi

```diff
@@ -553,10 +553,11 @@ PropertyMethodByNiceMonomorphism( IsSupersolvableGroup,
 InstallMethod(IsomorphismPermGroup,"via niceomorphisms",true,
   [IsGroup and IsFinite and IsHandledByNiceMonomorphism],
   # This is intended to be better than the generic ``action on element''
-  # method. However for example for matrix groups there are better methods.
+  # method (with requirement 'IsGroup and IsFinite').
+  # However for example for matrix groups there are better methods.
   # The downranking is compatible with that for the method for finite
   # matrix groups in 'lib/grpmat.gi'.
-  -NICE_FLAGS+5,
+  {} -> 1 - RankFilter( IsHandledByNiceMonomorphism ),
```

Comment:
There must be a cleaner way. The -NICE_FLAGS is intended to not get the upranking of `IsHandledBy`, but the 1 or 5 are pure guesses.

Reply:
I  should have been more clear: Is that the recommended way to adjust this?

Abstractly, what would be the recommended ranking to have a condition IsABC and IsDEF ignore the extra IsDEF ranking? While -RankFilter(IsDEF) seem right, it would ignore common implications and filters, i.e. if IsABC and IsDEF implies something extra. Should it rather be:

-RankFilter(IsABC and IsDEF)+RankFilter(IsABC),

though this is more kludgy.

Reply:
Yes what @hulpke writes un his last comment is the clean(est) way, as in general `RankFilter(A and B) > RankFilter(A) + RankFilter(B)`.

That said in many cases there are no common implications between A and B and it is unlikely that common implications will be added, and then we sometimes take the "lazy" (no offense intended) way... it is after all still cleaner than hard coded constants.

This is not intended to be an endorsement, though; it would of course still be best (but indeed a bit more cumbersome) to use the safe way described by Alexander

Reply:
Good idea.

From this point of view, the idea of an "incremental rank" makes the formulation so complicated.
What one wants to express is:
"Install the given method with requirements `[ x1, x2, ..., xn ]` but compute its (absolute) rank w.r.t. the requirements `[ x1', x2', ..., xn' ]`, plus an additional shift by a number `N`."

If we admit the format `[ [ x1', x2', ..., xn' ], N ]` for the `<rank>` argument in `InstallMethod` then `INSTALL_METHOD` can deal with the situation, by just taking the intended requirements for the rank computation. Then no zero argument function for computing the incremental rank is needed.

Reply:
I have the code for the change sketched above, and I want to add a description to the manual section "Applicable Methods and Method Selection". (Note that also the variant where the incremental rank is given via a function is not yet documented there.)